src/components/button.tsx

```diff
@@ -0,0 +1,22 @@
+import { twJoin } from "tailwind-merge";
+
+interface ButtonProps {
+  onClick?: () => void;
+  className?: string;
+  text?: string;
+  type?: "button" | "submit" | "reset" | undefined;
+}
```

Comment:
Instead of recreating the props of a button, why don't you use the native button types?
Also, instead of text, why don't you use children? Of course, if you want to limit the user to pass only a string, that's fine, otherwise you could just use children instead.
 This could be achieved doing something like this:
```suggestion
type ButtonProps = Pick<React.ButtonHTMLAttributes<HTMLButtonElement>, 'onClick' | 'className' | 'children' | 'type'>
```

Reply:
thats true, that way i could pass anyprops that's usually works inside a button.